### PR TITLE
Merged all GCI PRs related to DCE

### PR DESCRIPTION
**Pull request by saxenism**
I've merged all the Pull Requests from students of the Google Code-In program related to the "Dead Code Elimination" optimizer in my fork of the rco repository in a branch called expt_gci_dce_pr. The trivial changes in R/opt-dead-code.R were intentional(so that changes maybe detected and then upon merging in the main rco repo all the contributors could get credit).
Thank You :)

### CODE REVIEW

File: vignettes/opt-dead-code.Rmd [hunk 1]
```diff
@@ -30,6 +30,10 @@ speed_up <- function(obj) {
 
 # Dead Code Elimination
 
+## Idea
+
+Dead code elimination (also known as DCE, dead code removal, dead code stripping, or dead code strip) is an optimization to remove code which does not affect the program results. Removing such code has several benefits: it shrinks program size, an important consideration in some contexts, and it allows the running program to avoid executing irrelevant operations, which reduces its running time. It can also enable further optimizations by simplifying program structure. Dead code includes code that can never be executed (unreachable code), and code that only affects dead variables (written to, but never read again), that is, irrelevant to the program.
```
Comment (jcrodriguez1989) [inline vignettes/opt-dead-code.Rmd]: Oh, the idea behind the `Idea` section was to have an absolutely non-technical description of the optimizer.
this version is really similar to the `background` section.
that is why the student's PRs were so simplified. Please write something like "this optimizer will find chunks of code that would never be executed, or which execution won't influence the results...."

Comment (saxenism) [reply vignettes/opt-dead-code.Rmd]: Ohh!! I seemed to have missed the point ... 😅 ..
I'll fix it immediately...
Thanks :)

Comment (saxenism) [reply vignettes/opt-dead-code.Rmd]: Hello @jcrodriguez1989 , please have a look again. Now, I've used only the definitions used by the students and sewn them together to make it plausible as well as non-technical.
Thanks :D

Comment (hinduBale) [reply vignettes/opt-dead-code.Rmd]: Hello @jcrodriguez1989 , please have a look again. Now, I've used only the definitions used by the students and sewn them together to make it plausible as well as non-technical.
Thanks :)

**Review by jcrodriguez1989 — APPROVED** [review]

File: vignettes/opt-dead-code.Rmd [hunk 2]
```diff
@@ -30,6 +30,10 @@ speed_up <- function(obj) {
 
 # Dead Code Elimination
 
+## Idea
+
+Dead code are pieces of code that do not affect the output of a program. The removal of these unnecessary code lines can be defined as Dead Code Elimination. These code lines do not serve any purpose in the main program flow because these codes will never be executed mainly because the condition for their execution is logically impossible/infeasible. This removal results in perfomance enhancements.
```
Comment (jcrodriguez1989) [inline vignettes/opt-dead-code.Rmd]: typo 'perfomance' ~> 'performance'.

Comment (saxenism) [reply vignettes/opt-dead-code.Rmd]: Hi, I just fixed that typo, please have a look...
 Maybe I need a better text editor 😞

**Pull request merged**